firstvoices/backend/tests/test_apis/test_sites_api.py

```diff
@@ -257,6 +258,116 @@ def test_detail_banner_video(self):
+            widget=widget_two
+        )
+
+        site.homepage = widget_list
+        site.save()
+
+        response = self.client.get(f"{self.get_detail_endpoint(site.slug)}")
+        assert response.status_code == 200
+        response_data = json.loads(response.content)
+
+        assert len(response_data["homepage"]) == 2
+
+        response_widget_one = response_data["homepage"][0]
+        response_widget_two = response_data["homepage"][1]
+
+        assert response_widget_one == {
+            "id": str(widget_one.id),
+            "title": widget_one.title,
+            "widgetType": widget_one.widget_type,
+            "format": "Default",
+            "visibility": "Public",
+            "settings": [
+                {
+                    "key": widget_one_settings_one.key,
+                    "value": widget_one_settings_one.value,
+                },
+                {
+                    "key": widget_one_settings_two.key,
+                    "value": widget_one_settings_two.value,
+                },
+            ],
+        }
+        assert response_widget_two == {
+            "id": str(widget_two.id),
+            "title": widget_two.title,
+            "widgetType": widget_two.widget_type,
+            "format": "Default",
+            "visibility": "Public",
+            "settings": [
+                {
+                    "key": widget_two_settings_one.key,
+                    "value": widget_two_settings_one.value,
+                }
+            ],
+        }
+
+    @pytest.mark.parametrize(
+        "user_role, expected_visible_widgets",
+        [
+            (None, 1),
+            (Role.MEMBER, 2),
+            (Role.ASSISTANT, 3),
+            (Role.EDITOR, 3),
+            (Role.LANGUAGE_ADMIN, 3),
+        ],
+    )
+    @pytest.mark.django_db
+    def test_detail_homepage_permissions(self, user_role, expected_visible_widgets):
+        user = factories.UserFactory.create(id=1)
+        self.client.force_authenticate(user=user)
+
+        site = factories.SiteFactory.create(visibility=Visibility.PUBLIC)
+        if user_role is not None:
+            factories.MembershipFactory.create(user=user, site=site, role=user_role)
+
+        widget_list = factories.SiteWidgetListWithEachWidgetVisibilityFactory.create(
+            site=site
+        )
+        widget_public = widget_list.widgets.all()[0]
+        widget_members = widget_list.widgets.all()[1]
+        widget_team = widget_list.widgets.all()[2]
+        widget_public.site = site
+        widget_public.save()
+        widget_members.site = site
+        widget_members.save()
+        widget_team.site = site
+        widget_team.save()
+
+        site.homepage = widget_list
+        site.save()
+
+        response = self.client.get(f"{self.get_detail_endpoint(site.slug)}")
+        assert response.status_code == 200
+        response_data = json.loads(response.content)
+
+        assert len(response_data["homepage"]) == expected_visible_widgets
+
```

Comment:
I think we need a couple more test cases to cover your latest updates. A case where the widgets are not added to the list in order, to show that they are still serialized in order. And a case where a widget belongs to more than one list with different order values, to show that it is serialized in the correct order.